src/function.ts

```diff
@@ -140,6 +275,78 @@ abstract class FunctionBase<P, O>
       ResultSelector: "$.Payload",
     };
   }
+
+  protected static normalizeAsyncDestination<P, O>(
+    destination:
+      | FunctionAsyncOnSuccessDestination<P, O>
+      | FunctionAsyncOnFailureDestination<P>
+  ) {
+    return destination === undefined
+      ? undefined
+      : isEventBus<any>(destination)
+      ? new EventBridgeDestination(destination.bus)
+      : isFunction(destination)
+      ? new LambdaDestination(destination.resource)
+      : destination;
+  }
+
+  configureAsyncInvoke(config: EventInvokeConfigOptions<P, O>): void {
+    this.resource.configureAsyncInvoke({
+      ...config,
+      onSuccess: FunctionBase.normalizeAsyncDestination<P, O>(config.onSuccess),
+      onFailure: FunctionBase.normalizeAsyncDestination<P, O>(config.onFailure),
+    });
+  }
+
+  onSuccess(
```

Comment:
Does this inherit docs?

Reply:
No was just curious if it was happpening implicitly?

Reply:
Yeah, it inherits from the interface.